docs/docker/building-net-docker-images.md

```diff
@@ -0,0 +1,215 @@
+---
+title: Building .NET Core Docker Images
+description: Understanding Docker images and .NET Core
+keywords: .NET, .NET Core, Docker
+author: spboyer
+manager: wpickett
+ms.date: 08/29/2016
+ms.topic: article
+ms.prod: .net-core
+ms.technology: .net-core-technologies
+ms.devlang: dotnet
+ms.assetid: 03c28597-7e73-46d6-a9c3-f9cb55642739
+---
+
+
+#Building Docker Images for .NET Core Applications
+
+In order to get an understanding of how to use .NET Core and Docker together, we must first get to know the different Docker images that are offered and when is the right use case for them. Here we will walk through the variations offered, build an ASP.NET Core Web API, use the Yeoman Docker tools to create a debuggable container as well as peek at how Visual Studio Code can assist in the process. 
+
+## Docker Image Optimizations
+
+When building docker images for developers, we focused on 3 main scenarios
+
+- Images used to develop .NET Core apps
+- Images used to build .NET Core apps
+- Images used to run .NET Core apps
+
+Why 3 images?
+When developing, building and running containerized applications, we have different priorities.
+- **Development:**  How fast can you iterate changes, and the ability to debug the changes. The size of the image isn't as important, rather can you make changes to your code and see them quickly. Some of our tools, like [yo docker](https://aka.ms/yodocker) for use in VS Code use this image during development time. 
+- **Build:** What's needed to compile your app. This includes the compiler and any other dependencies to optimize the binaries. This image isn't the image you deploy, rather it's an image you use to build the content you place into a production image. This image would be used in your continuous integration, or build environment. For instance, rather than installing all the dependencies directly on a build agent, the build agent would instance a build image to compile the application with all the dependencies required to build the app contained within the image. Your build agent only needs to know how to run this docker image. 
+- **Prod:** How fast you can you deploy and start your image. This image is small so it can quickly travel across the network from your docker registry to your docker hosts. The contents are ready to run enabling the fastest time from docker run to processing results. In the immutable docker model, there's no need for dynamic compilation of code, nor razor files (`.cshtml`) as the contents of an image would never change. The content you place in this image would be limited to the binaries and content needed to run the applicaiton. For example, the published output using `dotnet publish` which contains the compiled binaries, images, .js and .css files. Over time, you'll see images that contain pre-jitted packages.   
```

Comment:
This raises a lot of ideas for other topics to cover later:
. Should this image contain minified, bundled JS and CSS?
. The lack of need for .cshtml implies that views have been pre compiled. Cool.
.